### PR TITLE
feat(perf): live perf audit + durable performance log

### PR DESCRIPTION
**Pull request by phrazzld**
## What
- Add `VoxPerfAudit` runner to measure **real STT + rewrite** latency for `raw/clean/polish` and emit JSON artifacts.
- Post a PR comment performance report on every PR (`.github/workflows/perf-audit.yml`).
- Persist durable JSON artifacts to `perf-audit` branch:
  - `audit/<sha>.json` for `master` pushes
  - `audit/pr/<pr>/<sha>.json` for PR runs (via `workflow_run` so secrets are available)
- Add optional runtime perf ingest (`VOX_PERF_INGEST_URL`) that uploads `pipeline_timing` events as NDJSON (best-effort, async, allowlisted).

## How To Run Locally
```bash
bash scripts/perf/make-fixture-audio.sh /tmp/vox-perf-fixture.caf
swift run VoxPerfAudit --audio /tmp/vox-perf-fixture.caf --output /tmp/vox-perf.json --iterations 2
python3 scripts/perf/format-perf-report.py --head /tmp/vox-perf.json
```

## CI Requirements
- Needs `OPENROUTER_API_KEY` and at least one STT key (`ELEVENLABS_API_KEY` / `DEEPGRAM_API_KEY`).
- Optional: `GEMINI_API_KEY` enables measuring Gemini-direct rewrite routing in CI.
- If secrets are unavailable (ex: fork PR), workflow posts a "Skipped" perf comment instead of failing.

## Privacy
- No transcript text or API keys; timings + sizes only.

## Summary by CodeRabbit

* **New Features**
  * Command‑line performance audit tool plus CI automation to run audits, post PR reports, and persist results.

* **New Features / Opt‑in**
  * Opt‑in runtime upload of pipeline timing events via VOX_PERF_INGEST_URL (disabled by default).

* **Documentation**
  * New performance docs and guides for audits, fixtures, CI workflows, and local runs.

* **Changes**
  * OpenAI Whisper removed as an STT provider; OPENAI_API_KEY removed from examples and UI/config surfaces.

## Before / After
Before: the PR was functional and mergeable but still had polish debt: unused provider metadata fields in the perf CLI result assembly, no explicit test coverage for unknown/case-insensitive forced STT provider values, no assertion that `onModelUsed` marks fallback usage, and perf docs lacked ingest payload schema plus explicit durable artifact retrieval steps.

After: removed unused metadata from `ResolvedProviders` in `VoxPerfAudit`, added edge-case tests for forced provider parsing/validation, added fallback-model callback behavior coverage in `OpenRouterClientTests`, and expanded performance docs with runtime NDJSON payload schema and `perf-audit` branch artifact retrieval instructions. Also opened follow-up architecture issue #280 to unify provider-resolution assembly between app runtime and perf CLI.

Screenshots: not applicable (internal refactor/tests/docs only; no UI surface changes).

### CODE REVIEW

File: .github/workflows/perf-audit-persist.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            sys.exit(0)
+          print(f"PR_NUMBER={pr}")
+          print(f"SHA={sha}")
+          PY
+
+          source meta.env
+          if [ -z "${PR_NUMBER:-}" ] || [ -z "${SHA:-}" ]; then
+            echo "Missing PR_NUMBER/SHA in head.json; skipping."
+            exit 0
+          fi
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply .github/workflows/perf-audit-persist.yml]: Addressed in e4b3f2b: added HEAD_JSON guard so persist step is skipped cleanly when audit was skipped.

File: scripts/perf/format-perf-report.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    lines: list[str] = []
+    lines.append("<!-- vox-perf-audit -->")
+    lines.append("## Performance Report")
+    lines.append("")
+    lines.append(f"- Head: `{short_sha(head_sha)}`")
+    if base:
+        lines.append(f"- Base: `{short_sha(base_sha)}`")
+    lines.append(f"- Providers: STT={stt_provider}, rewrite={rewrite_provider}")
+    lines.append(f"- Fixture: `{audio_file}` ({audio_bytes} bytes), iterations={iters} per level")
+    lines.append(f"- Note: paste stage is a no-op in CI; focus on generation timings.")
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply scripts/perf/format-perf-report.py]: Addressed in e4b3f2b: removed extraneous f-string prefix.

File: Sources/VoxPerfAudit/main.swift [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+                withIntermediateDirectories: true
+            )
+            try data.write(to: config.outputURL, options: [.atomic])
+        } catch {
+            if let err = error as? PerfAuditError {
+                fputs("\(err.description)\n", stderr)
+            } else {
+                fputs("\(error)\n", stderr)
+            }
+            exit(1)
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply Sources/VoxPerfAudit/main.swift]: Addressed in e4b3f2b: --help exits 0 (usage to stdout); invalid args exit 2.

File: Sources/VoxAppKit/PerformanceIngestClient.swift [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            flushCompleted(to: ingestURL)
+        }
+    }
+
+    private func flushCompleted(to ingestURL: URL) {
+        flushTask = nil
+        if !buffered.isEmpty {
+            scheduleFlush(to: ingestURL)
+        }
+    }
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply Sources/VoxAppKit/PerformanceIngestClient.swift]: Declining this one. Task closure here is created inside actor-isolated context and inherits that context, so the synchronous actor method call remains valid. Strict build with warnings-as-errors passes on this branch.

File: Sources/VoxProviders/OpenRouterClient.swift [hunk 5]
```diff
@@ -24,11 +27,13 @@ public final class OpenRouterClient: RewriteProvider {
             try Task.checkCancellation()
             let start = CFAbsoluteTimeGetCurrent()
             do {
+                let routerModel = currentModel.contains("/") ? currentModel : "google/\(currentModel)"
                 let result = try await executeRequest(
                     transcript: transcript,
                     systemPrompt: systemPrompt,
                     model: currentModel
                 )
+                onModelUsed?(routerModel, index > 0)
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply Sources/VoxProviders/OpenRouterClient.swift]: Fixed in 426b3f2. OpenRouter response decoding now reads top-level model and reports the served model in onModelUsed, with fallback to requested router model when response omits model. Added tests for both cases.

File: Tests/VoxPerfAuditKitTests/VoxPerfAuditKitTests.swift [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        #expect(d.p95 == 10)
+    }
+
+    @Test("Interpolates percentiles")
+    func test_percentiles_interpolate() {
+        let d = StageDistribution(samples: [0.0, 100.0])
+        #expect(d.p50 == 50)
+        #expect(d.p95 == 95)
+    }
+}
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply Tests/VoxPerfAuditKitTests/VoxPerfAuditKitTests.swift]: Fixed in 426b3f2. Renamed StageDistribution tests to follow the repo naming convention with explicit behavior/condition wording.

**Comment by phrazzld** [comment]
## PR Unblocked

**Conflicts**: none
**CI**: green (all required checks passing on 426b3f2)
**Reviews/Comments**: 6 addressed total — 3 fixed, 0 deferred, 3 declined with rationale in-thread

Ready for re-review.
